File: javascript/242_Valid_Anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    let dic = {};
    for (const c of s)
        dic[c] = (c in dic ? dic[c] + 1 : 1);
    for (const d of t) {
        if (d in dic) {
            if (--dic[d] === 0)
                delete dic[d];
        }
        else
            return false;
    }
    return Object.keys(dic).length === 0
};

File: python3/242_Valid_Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        dic = {}
        for a in s:
            dic[a] = dic[a] + 1 if a in dic else 1
        for d in t:
            if d in dic:
                dic[d] -= 1
                if dic[d] == 0:
                    del dic[d]
            else:
                return False
        return len(dic) == 0
